qualang_tools/units/units.py

```diff
@@ -0,0 +1,61 @@
+"""Tools to help handling units and conversions.
+
+Content:
+    - unit: class containing the relevant units.
+    - to_clock_cycles: converts durations from ns to clock cycles.
+    - demod2volts: converts demodulated data to volts.
+    - raw2volts: converts raw data to volts.
+"""
+
+
+class unit:
+    def __init__(self):
+        # Time units
+        self.ns = 1
+        self.us = 1000
+        self.ms = 1000_000
+        self.s = 1000_000_000
+        self.cc = 1 / 4
+
+        # Frequency units
+        self.mHz = 0.001
+        self.Hz = 1
+        self.kHz = 1000
+        self.MHz = 1000_000
+        self.GHz = 1000_000_000
+
+        # Voltage units
+        self.V = 1
+        self.mV = 1e-3
+        self.uV = 1e-6
+
+    # conversion functions
+    def to_clock_cycles(self, t):
+        """Converts a duration to clock cycles.
+
+        :param t: duration in ns.
+        :return: duration in clock cycles.
+        """
+        if not float(t).is_integer():
+            print(
+                f"Warning: the specified duration ({t}) to be converted to clock cycles in not an integer. It has been converted to int ({int(t)}) to avoid subsequent errors."
+            )
+            t = int(t)
+        return t // 4
+
+    def demod2volts(self, data, duration):

```

Comment:
Ain't there another factor of sqrt(2) or something?
Need to verify with a real OPX.

Reply:
I am not aware of that and I don't think it is on the documentation so please keep me updated

Reply:
I checked it and it seems you are right.
In the documentation, we have an extra factor of 2 which is not correct (but we do not know why).
Your function is good.